hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1984,4 +1984,19 @@ public static Optional<BucketCache> getBucketCacheFromCacheConfig(CacheConfig ca
     return Optional.empty();
   }
 
+  public List<String> getCachedFilesList() throws IOException {
+    List<String> tmp = new ArrayList<>();
+    File persistedFile = new File(persistencePath);
+    try (FileInputStream fis = new FileInputStream(persistedFile)) {
+      if (fis.skip(ProtobufMagic.lengthOfPBMagic()) != ProtobufMagic.lengthOfPBMagic()) {
+        throw new IOException(
+          "Incorrect number of bytes read while checking for protobuf magic number");
+      }
+      BucketCacheProtos.BucketCacheEntry proto =
+        BucketCacheProtos.BucketCacheEntry.parseDelimitedFrom(fis);
+      tmp.addAll(proto.getPrefetchedFilesMap().keySet());
+    }
+    return tmp;
+  }
```

Comment:
So this would make sense for a standalone tool, but since we are implementing this whole thing as part of the RS RPC API, why don't we simply return the in-memory contents?